### PR TITLE
Improve docs structure

### LINKED ISSUE #3493: Ensure that 2.0 docs are up-to-date and properly structured
### Which part of the project should be enhanced?

documentation

### Is your enhancement request related to a problem? Please describe.

_No response_

### Describe the solution you'd like

We need to check that the docs reflect the current state of 2.0 development and restructure them. `2.0.md` should be split up and removed.

### Describe alternatives you've considered

Have wrong/incomplete/confusing docs

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
documentation

### Description
First step to fix #3493

This PR restructures the existing documentation and introduces a "Design Documents" category.

Content is only changed where necessary for a coherent text, this is mostly moving stuff around and formatting

### Steps

- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: docs/content/dev/design_docs/api_auth.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+When a new token is requested via the private API, the backend generates a 64 bytes-long secret of
+cryptographically secure data and returns it as a base64url-encoded string, along with an identifier.
+That string can then be used by clients as a bearer token.
+
+A SHA-512 hash of the secret is stored in the database. To validate tokens, the backend computes the hash of the provided
+secret and checks it against the stored hash for the provided identifier.
+
+#### Choosing a hash function
+Unfortunately, there does not seem to be any explicit documentation about our exact use-case.
```
Comment (DerMolly) [inline docs/content/dev/design_docs/api_auth.md:19]:
```suggestion

Unfortunately, there does not seem to be any explicit documentation about our exact use-case.
```

File: docs/content/dev/design_docs/api_auth.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+An attacker gains read-only access to the database. Saving only hashes in the database prevents the attacker
+from authenticating themselves as a user. The hash-function does not need to be very slow,
+as the randomness of the original token prevents inverting the hash. The function actually needs to be reasonably fast,
+as the hash must be computed on every request to the public API.
+SHA-512 (or alternatively SHA3) fits this use-case.
+
+## Private API
+
+The private API uses a session cookie to authenticate the user.
+Sessions are handled using passport.js.
```
Comment (DerMolly) [inline docs/content/dev/design_docs/api_auth.md:37]:
```suggestion
Sessions are handled using [passport.js](https://www.passportjs.org/).
```

File: docs/content/dev/design_docs/events.md [hunk 3]
```diff
@@ -1,8 +1,12 @@
-In HedgeDoc we use an event system based on [EventEmitter2][eventemitter2]. It's used to reduce circular dependencies between different services and inform these services about changes.
+# Events
```
Comment (DerMolly) [inline docs/content/dev/design_docs/events.md:1]: The design docs disclaimer is missing

File: docs/content/dev/dev_notes.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Development Notes
```
Comment (DerMolly) [inline docs/content/dev/dev_notes.md:1]:
```suggestion
# Development Notes

```

File: docs/content/dev/dev_notes.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+This document collects notes and decisions taken during the development of HedgeDoc 2.0.
+It should be converted to a properly structured documentation, but having unstructured docs
+is better than having no docs.
+
+## Supported databases
```
Comment (DerMolly) [inline docs/content/dev/dev_notes.md:6]:
```suggestion
## Supported databases

```

**Review by DerMolly — CHANGES_REQUESTED** [review]

**Review by InnayTool — APPROVED** [review]

**Review by DerMolly — APPROVED** [review]

File: docs/content/dev/design_docs/api_auth.md (hunk 1 above)
Comment (ErikMichelson) [reply docs/content/dev/design_docs/api_auth.md:25]: What should be changed here? You introduce a new blank line, resulting in two blank lines being below the heading...

File: docs/content/dev/design_docs/api_auth.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+feature. It is not a user guide and may or may not be fully implemented.
+
+## Public API
+
+All requests to the public API require authentication using a [bearer token](https://datatracker.ietf.org/doc/html/rfc6750).
+
+This token can be generated using the profile page in the frontend
+(which in turn uses the private API to generate the token).
+
+### Token Generation
```
Comment (ErikMichelson) [inline docs/content/dev/design_docs/api_auth.md:14]:
```suggestion
### Token generation
```

**Review by InnayTool — APPROVED** [review]

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**